src/main/scala/ch/epfl/bluebrain/nexus/kg/async/Projects.scala

```diff
@@ -177,6 +210,44 @@ object Projects {
         case g @ GetSuccess(LWWRegisterKey(_), _) => g.get(viewKey(ref)).value.value
         case NotFound(_, _)                       => Set.empty
       }
+
+    override def addView(ref: ProjectRef, view: View, updateRev: Boolean): Future[Boolean] = {
+      val found = (v: View) =>
+        if (updateRev) v.id == view.id && v.rev >= view.rev
+        else v.id == view.id
+
+      views(ref).flatMap { viewSet =>
+        if (viewSet.exists(found)) Future.successful(false)
+        else {
+          val empty  = LWWRegister(TimestampedValue(0L, Set.empty[View]))
+          val value  = TimestampedValue(view.instant.toEpochMilli, viewSet + view)
+          val update = Update(viewKey(ref), empty, WriteMajority(tm.duration))(_.withValue(value))
+          (replicator ? update).flatMap {
+            case UpdateSuccess(LWWRegisterKey(_), _) =>
```

Comment:
partial function is the same in `addView` and `removeView`. Refactor?

Reply:
to be honest, this sort of "refactoring" makes the code harder to understand, but alright...